service/rpccommon/server.go

```diff
@@ -146,7 +145,6 @@ func (s *ServerImpl) Run() error {
 	suitableMethods(rpcServer, s.methodMaps[1], s.log)
 
 	go func() {
-		defer s.listener.Close()
```

Comment:
I think the erroneous `.Close()` would be in the `Stop()` method. I believe this is the correct place to close the listener.